kore/app/exec/Main.hs

```diff
@@ -563,6 +563,11 @@ main = do
             Main.exeName
             (parseKoreExecOptions startTime)
             parserInfoModifiers
+    Foldable.for_ (localOptions options >>= smtPrelude) (\filePath ->
+        Monad.whenM
+            (not <$> doesFileExist filePath)
+            (error $ "SMT prelude file does not exist: " <> filePath)
+        )
```

Comment:
Let's move this check into `mainWithOptions` and avoid calling `localOptions` twice.